src/app/features/swap-form/reducer.ts

```diff
@@ -0,0 +1,123 @@
+    // user-inputted notional amount
+    notionalAmount: string;
+  };
+  // State of cashflow calculator
+  cashflowCalculator: {
+    // user-inputted predicted variable apy
+    predictedApy: string;
+    // cached variable factor from termStart to now
+    variableFactorStartNow: number;
+    // Additional cashflow resulted from prospective swap, form now to termEnd.
+    additionalCashflow: string;
+    // Total cashflow resulted from past and prospective swaps, from termStart to termEnd.
+    totalCashflow: string;
+    // Flag that signals whether cashflow calculator was initialised
+    initialised: boolean;
+    // Flag that signals potential errors in cashflow calculation
+    error: boolean;
+  };
+};
+
+const initialState: SliceState = {
+  amm: undefined,
+  position: undefined,
+  prospectiveSwap: {
+    notionalAmount: '0',
+  },
+  cashflowCalculator: {
+    predictedApy: '0',
+    variableFactorStartNow: 0,
+    additionalCashflow: '+0.00',
+    totalCashflow: '+0.00',
+    initialised: false,
+    error: false,
+  },
+};
+
+export const slice = createSlice({
+  name: 'swapForm',
+  initialState,
+  reducers: {
+    setNotionalAmountAction: (
+      state,
+      {
+        payload: { value },
+      }: PayloadAction<{
+        value: string;
+      }>,
+    ) => {
+      state.prospectiveSwap.notionalAmount = value;
+    },
+    setPredictedApyAction: (
+      state,
+      {
+        payload: { value },
+      }: PayloadAction<{
+        value: string;
+      }>,
+    ) => {
+      state.cashflowCalculator.predictedApy = value;
+      if (parseFloat(toUSFormat(value) as string) < 0) {
+        state.cashflowCalculator.error = true;
+      } else {
+        state.cashflowCalculator.error = false;
+      }
+    },
+    refreshCashflows: (
+      state,
+      {
+        payload: { amm },
+      }: PayloadAction<{
+        amm: AMM; //TODO Alex: remove amm and use amm from state
+      }>,
+    ) => {
+      if (state.cashflowCalculator.error) {
+        return;
+      }
+
+      const { additionalCashflow, totalCashflow } = amm.getExpectedCashflowInfo({
+        position: state.position as Position | undefined, //TODO Alex
+        fixedTokenDeltaBalance: Number(state.prospectiveSwap.notionalAmount), //TODO Alex
+        variableTokenDeltaBalance: -Number(state.prospectiveSwap.notionalAmount), //TODO Alex
+        variableFactorStartNow: state.cashflowCalculator.variableFactorStartNow,
+        predictedVariableApy: Number(state.cashflowCalculator.predictedApy), //TODO Alex
+      });
+
+      state.cashflowCalculator.additionalCashflow = formatCurrency(
+        additionalCashflow,
+        true,
+        true,
+        2,
+        4,
+      );
+
+      state.cashflowCalculator.totalCashflow = formatCurrency(totalCashflow, true, true, 2, 4);
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(initialiseCashflowCalculator.pending, (state, {}) => {
+        state.cashflowCalculator.initialised = false;
```

Comment:
to be honest not sure why you don't use the 'idle' | 'success' | 'pending' | 'error' instead of boolean, for the initialised. Right now it is not possible to answer the question is it loading or initial state.

Reply:
Addressed [6b0d2f3](https://github.com/Voltz-Protocol/voltz-ui/pull/570/commits/6b0d2f3843234382bf7a2334f42b64cb5096adce)